src/codegen/llvm.zig

```diff
@@ -11339,6 +11343,25 @@ const ParamTypeIterator = struct {
                     return .byref;
                 }
             },
+            .Interrupt => {
+                switch (target.cpu.arch) {
+                    .x86 => switch (it.zig_index) {
+                        0 => if (ty.zigTypeTag(mod) != .Pointer) @panic("first argument must be a pointer"),
+                        1 => if (ty.zigTypeTag(mod) != .Int or ty.bitSize(mod) != 32) @panic("second argument must be a u32"),
+                        else => @panic("too many arguments"),
+                    },
+                    .x86_64 => switch (it.zig_index) {
+                        0 => if (ty.zigTypeTag(mod) != .Pointer) @panic("first argument must be a pointer"),
+                        1 => if (ty.zigTypeTag(mod) != .Int or ty.bitSize(mod) != 64) @panic("second argument must be a u64"),
+                        else => @panic("too many arguments"),
+                    },
+                    else => @panic("TODO: handle Interrupt callconv parameters for other architectures."),
+                }
```

Comment:
These checks should happen in `Sema.funcCommon` and should result in compile errors instead of panics.

If you're interested in implementing the checks for other architectures you can find what clang does for them here:
https://github.com/llvm/llvm-project/blob/release/18.x/clang/lib/Sema/SemaDeclAttr.cpp#L7455

Reply:
Moved the checks into `Sema.zig`. It seems like the other architectures accept parameters in the attribute itself, as opposed to the function signature. I think we could make this work in zig using builtins or checking for special comptime parameters, but that's out of scope for this PR.